src/state/apis/swapper/swapperApi.ts

```diff
@@ -0,0 +1,93 @@
+import { createApi, fakeBaseQuery } from '@reduxjs/toolkit/dist/query/react'
+import { AssetId, ChainId } from '@shapeshiftoss/caip'
+import { GetTradeQuoteInput, Swapper, TradeQuote } from '@shapeshiftoss/swapper'
+import { getSwapperManager } from 'components/Trade/hooks/useSwapper/swapperManager'
+import { AssetsState } from 'state/slices/assetsSlice/assetsSlice'
+import { FeatureFlags, Preferences } from 'state/slices/preferencesSlice/preferencesSlice'
+
+type GetUsdRateArgs = {
+  rateAssetId: AssetId | undefined
+  buyAssetId: AssetId | undefined
+  sellAssetId: AssetId | undefined
+}
+
+type GetUsdRateReturn = {
+  usdRate: string
+}
+
+type State = {
+  assets: AssetsState
+  preferences: Preferences
+}
+
+const getBestSwapperFromArgs = async (
+  buyAssetId: AssetId | undefined,
+  sellAssetId: AssetId | undefined,
+  featureFlags: FeatureFlags,
+): Promise<Swapper<ChainId>> => {
+  if (!buyAssetId) throw new Error('buyAssetId is undefined')
+  if (!sellAssetId) throw new Error('sellAssetId is undefined')
+  const swapperManager = await getSwapperManager(featureFlags)
+  const swapper = await swapperManager.getBestSwapper({
+    buyAssetId,
+    sellAssetId,
+  })
+  if (!swapper) throw new Error('swapper is undefined')
+  return swapper
+}
+
+export const swapperApi = createApi({
+  reducerPath: 'swapperApi',
+  baseQuery: fakeBaseQuery(),
+  // refetch if network connection is dropped, useful for mobile
+  refetchOnReconnect: true,
+  keepUnusedDataFor: 60,
+  endpoints: build => ({
+    getUsdRate: build.query<GetUsdRateReturn, GetUsdRateArgs>({
+      queryFn: async ({ rateAssetId, buyAssetId, sellAssetId }, { getState }) => {
+        const state: State = getState() as unknown as State // ReduxState causes circular dependency
+        const {
+          assets,
+          preferences: { featureFlags },
+        } = state
+        try {
+          if (!rateAssetId) throw new Error('rateAssetId is undefined')
+          const swapper = await getBestSwapperFromArgs(buyAssetId, sellAssetId, featureFlags)
+          const rateAsset = assets.byId[rateAssetId]
+          const usdRate = await swapper.getUsdRate(rateAsset)
+          const data = { usdRate }
+          return { data }
+        } catch (e) {
+          const data = `getAssetDescription: error fetching usd rate for ${rateAssetId}`
+          const status = 400
+          const error = { data, status }
+          return { error }
+        }
+      },
+    }),
+    getTradeQuote: build.query<TradeQuote<ChainId>, GetTradeQuoteInput>({
```

Comment:
Would still abstract this to its own type for consistency
```suggestion
    getTradeQuote: build.query<GetTradeQuoteOutput, GetTradeQuoteInput>({
```

Reply:
For sure - fixed up in https://github.com/shapeshift/web/pull/2526/commits/00a8dc2ce49c18d6b93905b234cf52fef91ff491.